sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -525,17 +544,187 @@ def generate_queue_url(event):
+
+    Parameters
+    ----------
+    session : sqlalchemy.orm.Session
+        Database session.
+    file_obj : ScienceFilePath
+        The file object for which to calculate the date range.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the start date and end date in the format YYYYMMDD.
+    """
+    # Look for latest and greatest file_path in the repointing table.
+    latest_repoint_file = (
+        session.query(models.RepointTable)
+        .order_by(desc(models.RepointTable.file_path))
+        .first()
+    )
+    if not latest_repoint_file:
+        raise ValueError("No repoint file found in the database.")
+
+    latest_repoint_file = latest_repoint_file.file_path
+
+    logger.info(
+        "Latest repoint file used to calculate date range"
+        f"for ENA and GLOWS instruments - {latest_repoint_file}"
+    )
+
+    repoint_path = imap_data_access.download(latest_repoint_file)
+    repoint_df = pd.read_csv(repoint_path)
+    # Set start date to be repoint_end_utc of i_pointing from the input file.
+    # Set end date to be repoint_end_utc of i_pointing + 1.
+    # Will there be a case when i_pointing + 1 is not in the
+    # repoint file?
+    #   No. Because WebPODA takes care of this. It makes sure that the L0 data
+    #   has i_pointing + 1 in the repoint file.
+    start_date = repoint_df.loc[
+        repoint_df["repoint_id"] == file_obj.repointing, "repoint_start_utc"
+    ].iloc[0]
+    start_date = datetime.datetime.strptime(
+        start_date, "%Y-%m-%dT%H:%M:%S.%f"
+    ).strftime("%Y%m%d")
+    end_date = repoint_df.loc[
+        repoint_df["repoint_id"] == file_obj.repointing + 1, "repoint_end_utc"
+    ].iloc[0]
+    end_date = datetime.datetime.strptime(end_date, "%Y-%m-%dT%H:%M:%S.%f").strftime(
+        "%Y%m%d"
+    )
+    return start_date, end_date
+
+
+def determine_date_range(session, file_obj):
+    """Determine the start and end dates based on the file type.
+
+    This date range is used to query upstream dependencies for the file.
+
+    Parameters
+    ----------
+    session : sqlalchemy.orm.Session
+        Database session.
+    file_obj : SPICEFilePath, ScienceFilePath, or AncillaryFilePath
+        The file object for which to determine the date range.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the start date and end date in the format YYYYMMDD.
+    """
+    if isinstance(file_obj, SPICEFilePath):
+        # TODO: fix date range if/when repoint file ingestion event is
+        # passed to batch starter to kickoff HARD or SOFT_TRIGGER downstream jobs.
+        # Convert datetime object to string of format YYYYMMDD
+        file_type = file_obj.spice_metadata["type"]
+        if file_type == "repoint":
+            # Get last two repoint files from the database.
+            # Then calculate the date range based on those two files.
+            start_date, end_date = calculate_pointing_date_range(session)
```

Comment:
Based on discussion with Greg on zoom, we are changing what database to query for this. We are using the new table discussed above. Query spice_table and query for pointing_attitude type and get date range from that? If we do that, DPS file format is this:
>    dps_file_pattern = (
        r"(?P<type>imap_dps)_"
        r"(?P<start_year_doy>\d{4}_\d{3})_"
        r"(?P<end_year_doy>[\d]{4}_[\d]{3})_"
        r"(?P<version>\d+)\."
        r"(?P<extension>ah\.bc)"
    )

we query for the latest and greatest dps file, then parse and get end_date of that latest file. that end date becomes the start date to query repoint file. all pointing records that have pointing_start_utc >=end_date of the last DPS file.

```suggestion
            # Get last two repoint files from the database.
            # Then calculate the date range based on those two files.
            start_date, end_date = calculate_pointing_date_range(session)
```